client/blocks/disconnect-jetpack/index.jsx

```diff
@@ -305,20 +312,14 @@ export default connect(
 			rewindState: rewindState.state,
 		};
 	},
-	dispatch => ( {
-		setAllSitesSelected: () => dispatch( setAllSitesSelected ),
-		recordGoogleEvent: () => dispatch( recordGoogleEventAction ),
-		recordTracksEvent: () => dispatch( recordTracksEventAction ),
-		disconnect: () => dispatch( disconnect ),
-		successNotice: () => dispatch( successNotice ),
-		errorNotice: () => dispatch( errorNotice ),
-		infoNotice: () => dispatch( infoNotice ),
-		removeNotice: () => dispatch( removeNotice ),
-		trackTryRewind: siteSlug => {
-			dispatch( recordTracksEventAction( 'calypso_disconnect_jetpack_try_rewind' ) );
-			page.redirect( `/stats/activity/${ siteSlug }` );
-		},
-		trackTryRewindHelp: () =>
-			dispatch( recordTracksEventAction( 'calypso_disconnect_jetpack_try_rewind_help' ) ),
-	} )
+	{
+		setAllSitesSelected,
+		recordGoogleEvent: recordGoogleEventAction,
+		recordTracksEvent: recordTracksEventAction,
+		disconnect,
+		successNotice,
+		errorNotice,
+		infoNotice,
+		removeNotice,
+	}
```

Comment:
I don't think it's necessary to roll all this back. It should be enough just passing the missing arguments, like modifying
```
disconnect: () => dispatch( disconnect ),
```
to
```
disconnect: ( siteId ) => dispatch( disconnect( siteId ) ),
```
I would actually prefer if these weren't modified so we don't have to put more complex functions like `handleTryRewind` and `trackTryRewindHelp` in the component implementation and we could leave them in the connect area.

Reply:
Sure, feel free to tweak this.

Reply:
> I don't think it's necessary to roll all this back. It should be enough just passing the missing arguments, like modifying
> ```
> disconnect: () => dispatch( disconnect ),
> ```
> to
> ```
> disconnect: ( siteId ) => dispatch( disconnect( siteId ) ),
> ```

This doesn't really make any difference, and it's equivalent to

```
disconnect
```

which is more elegant, and the kind of notation we try to use throughout Calypso.

> I would actually prefer if these weren't modified so we don't have to put more complex functions like `handleTryRewind` and `trackTryRewindHelp` in the component implementation and we could leave them in the connect area.

That's a fair point. I'll point out that prior to #22724, the component was using class methods to trigger tracks events -- arguably, #22724 made it somewhat inconsistent in that regard.

If you'd like to keep those functions in `connect`, it's still possible to do so without invoking `dispatch`, e.g.

```
{
/*...*/
trackTryRewindHelp: () => recordTracksEventAction( 'calypso_disconnect_jetpack_try_rewind_help' )
}
```

The 0-ary function on the right-hand side will then be auto-bound (wrapped) with `dispatch` by `connect`.

It's a little trickier but still possible for `trackTryRewind`, I'll follow up on that.

Reply:
> It's a little trickier but still possible for `trackTryRewind`, I'll follow up on that.

As discussed via DM, it should be possible using [`withAnalytics`](https://github.com/Automattic/wp-calypso/tree/master/client/state/analytics#usage) and  [`navigate` ](https://github.com/Automattic/wp-calypso/tree/master/client/state/navigation) 🙂